src/fty_config_manager.cc

```
@@ -0,0 +1,448 @@
+#include <sstream>
+#include <vector>
+
+#include "fty_config_classes.h"
+
+
+using namespace std::placeholders;
+
+namespace config
+{   
+    /**
+     * Constructor
+     * @param parameters
+     * @param streamPublisher
+     */
+    ConfigurationManager::ConfigurationManager(const std::map<std::string, std::string> & parameters)
+    : m_parameters(parameters), m_aug(NULL)
+    {
+        init();
+    }
+
+    /**
+     * Destructor
+     */
+    ConfigurationManager::~ConfigurationManager()
+    {
+        log_debug("Release all config resources");
+        if (!m_aug)
+        {
+            aug_close(m_aug);
+            log_debug("Augeas resource released");
+        }
+        if (m_msgBus) 
+        {
+            delete m_msgBus;
+            log_debug("Message bus resource released");
+        }
+    }
+
+    /**
+     * Initialization class
+     */
+    void ConfigurationManager::init()
+    {
+        try
+        {
+            // Augeas tool init
+            int augeasOpt = getAugeasFlags(m_parameters.at(AUGEAS_OPTIONS));
+            log_debug("augeas options: %d", augeasOpt);
+
+            m_aug = aug_init(FILE_SEPARATOR, m_parameters.at(AUGEAS_LENS_PATH).c_str(), augeasOpt);
+            if (!m_aug)
+            {
+                throw ConfigurationException("Augeas tool initialization failed");
+            }
+
+            // Message bus init
+            m_msgBus = messagebus::MlmMessageBus(m_parameters.at(ENDPOINT_KEY), m_parameters.at(AGENT_NAME_KEY));
+            m_msgBus->connect();
+            
+            // Listen all incoming request
+            auto fct = std::bind(&ConfigurationManager::handleRequest, this, _1);
+            m_msgBus->receive(m_parameters.at(QUEUE_NAME_KEY), fct);
+        }
+        catch (messagebus::MessageBusException& ex)
+        {
+            log_error("Message bus error: %s", ex.what());
+        } catch (...)
+        {
+            log_error("Unexpected error: unknown");
+        }
```

Silently eating errors. Let exceptions bubble up.

---

Agree.